### PR TITLE
Working install v0.1

### PR DESCRIPTION
**Pull request by SiM22**
I know there is already a very similar PR but below I will list changes and features added.  Have renamed all the comments/directories that mention `paperless-ng` to `paperless-nix`

* Renamed all the comments/directories that mention `paperless-ng` to `paperless-ngx` [Note: Variables are still not updated to ngx
* Added paperless export directory
* Added NFS mount possibilities for external consumption directory [if using an LXC container please make the relevant changes to allow the LXC to use NFS]
* Added Tika/Gotenburg endpoint variable
* Added step to upgrade pip within virtualenv prior to running the install of requirements [This is because if the step to upgrade]

Currently working with v1.8.0

### CODE REVIEW

**Comment by SiM22** [comment]
@shamoon @qcasey any chance someone can take a look? 

I do not mind being an active member on the ansible repo if you need more eyes on this side of the fence

**Comment by shamoon** [comment]
Appreciate the PR. The reason we separated out the ansible stuff was because we don’t have a lot of expertise on the dev team with it. For myself for example I am of almost no use for this. If you’d like to become the go-to person for this that might make sense, qcasey I believe there really isn’t anyone else working on this at the moment correct?

**Comment by stumpylog** [comment]
I actually use Ansible to configure and maintain my own systems, though I don't deploy paperless with it (I dockerize everything possible).  So I'll have a look through.

**Comment by qcasey** [comment]
> qcasey I believe there really isn’t anyone else working on this at the moment correct?

No one is maintaining this as far as I know. Just #2 which this claims to be an extension of anyway.

stumpylog if this tests OK and @SiM22 wants to maintain I think that would be a good solution.

File: tasks/main.yml [hunk 1]
```diff
@@ -254,18 +252,27 @@
...
+    mode: "750"
+  when: consumption_mount_exist == false
 
 - name: rename initial config
   command:
     cmd: "mv -f {{ paperlessng_directory }}/paperless.conf {{ paperlessng_directory }}/paperless.conf.template"
     removes: "{{ paperlessng_directory }}/paperless.conf"
 
-- name: configure paperless-ng
+- name: configure paperless-ngx
```
Comment (stumpylog) [inline tasks/main.yml:275]: This fails with the following vars undefined:
- paperlessng_url
- paperlessng_enable_update_check

File: tasks/main.yml [hunk 2]
```diff
@@ -105,79 +123,58 @@
...
+
+- name: register current existing state
+  set_fact:
+    update_installation: '{{ paperlessng_current_commit.stdout != paperlessng_commit | string }}'
     reconfigure_only: "{{ paperlessng_current_commit.stdout == paperlessng_commit | string }}"
+  when: not fresh_installation
 
 - block:
-    - name: backup current paperless-ng installation
+    - name: backup current paperless-ngx installation
```
Comment (stumpylog) [inline tasks/main.yml:177]: If `/opt/paperless-ngx` doesn't already exist, this step fails.

File: tasks/main.yml [hunk 3]
```diff
@@ -254,18 +252,27 @@
...
 
 - name: rename initial config
   command:
     cmd: "mv -f {{ paperlessng_directory }}/paperless.conf {{ paperlessng_directory }}/paperless.conf.template"
     removes: "{{ paperlessng_directory }}/paperless.conf"
 
-- name: configure paperless-ng
+- name: configure paperless-ngx
   lineinfile:
     path: "{{ paperlessng_directory }}/paperless.conf.template"
```
Comment (stumpylog) [inline tasks/main.yml:277]: This file doesn't exist.  It appears to be skipping any installation if nothing exists already.

**Review by stumpylog — CHANGES_REQUESTED** [review]
I tried to give this branch a run in a VM.  Although it does get somewhat into the setup, I'm not able to run it to completion just yet.

File: README.md [hunk 4]
```diff
@@ -1,92 +1,92 @@
 ### :warning: THIS REPO IS UNTESTED! Ansible is not currently supported by paperless-ngx. Use only at your own peril. :warning:
```
[the remark this answers is not in the document]
Comment (qcasey) [reply README.md:1]: Agreed, the warning can be removed.

**Comment by SiM22** [comment]
ok let me get some of those things fixed, this is probably related to me using a copy paste technique from the one i got running

File: README.md [hunk 5]
```diff
@@ -1,92 +1,92 @@
...
-Role Variables
---------------
-
-Most configuration variables from paperless-ng itself are available and accept their respective arguments.
+```
+- hosts: all
+  roles:
+  - role: paperless-ngx
+    become: yes
+    Role Variables
```
[the remark this answers is not in the document]
Comment (SiM22) [reply README.md]: Fixed

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (SiM22) [reply README.md]: Removed warning

File: README.md [hunk 6]
```diff
@@ -1,92 +1,92 @@
...
-To install a specific version of paperless-ng, use the tag name of the release, e. g. `ng-1.4.4`, or specify a branch or commit id.
-
-    paperlessng_redis_host: localhost
-    paperlessng_redis_port: 6379
 
+Additional variables available in this role are listed below, along with default values:
+```paperlessng_version: latest```
+The [release](https://github.com/paperless-ngx/paperless-ngx/releases) archive version of paperless-ngx to install.
+`latest` stands for the latest release of paperless-ngx.
+To install a specific version of paperless-ngx, use the tag name of the release, e. g. `1.4.4` when the tag is `ngx-1.4.4`.
```
[the remark this answers is not in the document]
Comment (SiM22) [reply README.md]: default set to 1.8.0

File: README.md [hunk 7]
```diff
@@ -1,92 +1,92 @@
...
-    paperlessng_version: latest
 
-The [release](https://github.com/jonaswinkler/paperless-ng/releases) archive version of paperless-ng to install.
-`latest` stands for the latest release of paperless-ng.
-To install a specific version of paperless-ng, use the tag name of the release, e. g. `ng-1.4.4`, or specify a branch or commit id.
-
-    paperlessng_redis_host: localhost
-    paperlessng_redis_port: 6379
 
+Additional variables available in this role are listed below, along with default values:
```
[the remark this answers is not in the document]
Comment (SiM22) [reply README.md:28]: added as a table. probably could be prettier

File: tasks/main.yml [hunk 8]
```diff
@@ -511,20 +533,81 @@
...
+    state: absent
+  when: paperlessng_redis_host != 'localhost' or paperlessng_redis_host != '127.0.0.1'
+
+- name: NFS utility present for redhat-like
+  ansible.builtin.yum:
+    name:
+      - nfs-utils
+      - nfs4-acl-tools
+    state: present
+  when: ansible_os_family == "RedHat" and paperlessng_consumption_dir_nfs_enabled and paperlessng_is_lxc != "True"
```
[the remark this answers is not in the document]
Comment (SiM22) [reply tasks/main.yml]: added as list

File: tasks/main.yml [hunk 9]
```diff
@@ -4,11 +4,26 @@
...
+  apt:
+    upgrade: yes
+
+- name: Check and set fact if consumption mount point exists
+  set_fact:
+    consumption_mount_exist: true
+  when: item.mount == '{{ paperlessng_consumption_dir }}'
+  loop: "{{ ansible_facts.mounts }}"
+
+- name: Show if consumption mount exists
```
[the remark this answers is not in the document]
Comment (SiM22) [reply tasks/main.yml]: removed as no idea why i had it added previously. May just be a case of working through errors and putting stuff in. But not required to get role to deploy

File: tasks/main.yml [hunk 10]
```diff
@@ -105,79 +123,58 @@
...
-          when: commit.status == 200
-        - name: fail
-          fail:
-            msg: "Can not determine commit from `paperlessng_version=={{ paperlessng_version }}`!"
-          when: commit.status != 200
-      when: branch.status == 404
-  when: not(paperlessng_version | regex_search("^(ng-)?(\d+\.\d+\.\d+)$"))
-
-- name: check for paperless-ng installation
+        paperlessng_commit: "{{ commit.json | json_query('sha') }}"
```
[the remark this answers is not in the document]
Comment (SiM22) [reply tasks/main.yml:133]: Added as a dependency in the Readme

File: tasks/main.yml (hunk 1 above)
Comment (SiM22) [reply tasks/main.yml:257]: Added both into defaults, missed out initially as was part of my group_vars.

File: tasks/main.yml (hunk 2 above)
Comment (SiM22) [reply tasks/main.yml:167]: The flags were wrong on the `register current state` step. I was testing the upgrade to 1.8.0. Both flags now set to follow the fresh installation path

File: tasks/main.yml (hunk 3 above)
Comment (SiM22) [reply tasks/main.yml:259]: Same as above. The flags were wrong on the register current state step. I was testing the upgrade to 1.8.0. Both flags now set to follow the fresh installation path

File: README.md [hunk 11]
```diff
@@ -1,92 +1,92 @@
...
+paperlessng_directory: /opt/paperless-ngx
+```
+Root directory paperless-ngx is installed into.
+```
+paperlessng_virtualenv: "{{ paperlessng_directory }}/.venv"
+```
+Directory used for the virtual environment for paperless-ngx.
+```
+paperlessng_ocr_languages:
+- eng
```
[the remark this answers is not in the document]
Comment (SiM22) [reply README.md]: This is now part of the table

**Comment by SiM22** [comment]
ok, I believe I have addressed all the comments. Most of them were due to me getting the role to work and then adding to it. A fresh set of eyes has helped get it a bit more tidy :-)

**Review by stumpylog — APPROVED** [review]
I made some changes directly, with which I was able to reach the end of my playbook and re-run it without a failure.  

There's more which can be cleaned up in the future if people want to tackle it, but this works for me as is.

**Pull request merged**